### PR TITLE
Add db manager tests for "rename" functions

### PR DESCRIPTION
**Pull request by shati-patel**
Follow-up to https://github.com/github/vscode-codeql/pull/1928 to add some "integration" tests (specifically testing the integration between the db manager and db config store).

These tests check renaming lists and local databases.

## Checklist

N/A—internal only 🎭

- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

**Review by charisk — APPROVED** [review]
LGTM, just some suggestions.

File: extensions/ql-vscode/test/unit-tests/databases/db-manager.test.ts [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    await remove(tempWorkspaceStoragePath);
+  });
+
+  const dbConfigFilePath = join(
+    tempWorkspaceStoragePath,
+    "workspace-databases.json",
+  );
+
+  describe("renaming items", () => {
+    it("should rename remote db list", async () => {
```
[the remark this answers is not in the document]
Comment (shati-patel) [reply extensions/ql-vscode/test/unit-tests/databases/db-manager.test.ts:47]: Not a particularly good reason 😂 I sort of broke down the renaming functionality into 4 cases: 
- remote lists
- local lists
- local "top-level" db
- local db inside a list

And then basically added one test for each! Happy to go with a different breakdown though 😀

**Comment by shati-patel** [comment]
> LGTM, just some suggestions.

Thanks for the review! I've added some more cases to test that the name hasn't changed 👍🏽  
(There's probably more tidy-up/restructuring to be done, but I'm going to leave that rabbit hole for now... 🐰 🕳️ )

File: extensions/ql-vscode/test/unit-tests/databases/db-manager.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (charisk) [reply extensions/ql-vscode/test/unit-tests/databases/db-manager.test.ts:57]: Sorry I forgot that's it's not possible to rename a remote db 🤦

**Review by charisk — APPROVED** [review]

**Comment by shati-patel** [comment]
Hmm, I think the eternal "pending" unit test check is a legitimate problem -- will investigate 🔍

File: extensions/ql-vscode/test/unit-tests/databases/db-manager.test.ts [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+      });
+
+      // Check that the local list has not been renamed
+      const localLists = dbConfigFileContents.databases.local.lists;
+      expect(localLists.length).toBe(1);
+      expect(localLists[0]).toEqual({
+        name: "my-list-1",
+        databases: [localDb],
+      });
+    });
```
Comment (charisk) [inline extensions/ql-vscode/test/unit-tests/databases/db-manager.test.ts:92]: We should also do `dbConfigStore.dispose()` at the end of each test (ideally in a `afterEach()`, with `beforeEach` creating the objects).

**Pull request merged**